packages/core/contracts/financial-templates/common/financial-product-libraries/long-short-pair-libraries/LinearLongShortPairFinancialProductLibrary.sol

```diff
@@ -32,47 +32,48 @@ contract LinearLongShortPairFinancialProductLibrary is LongShortPairFinancialPro
+     * @param longShortPair address of the LSP contract.
      * @param upperBound the upper price that the linear LSP will operate within.
      * @param lowerBound the lower price that the linear LSP will operate within.
      * @dev Note: a) Any address can set these parameters b) existing LSP parameters for address not set.
      * c) upperBound > lowerBound.
-     * d) parameters price can only be set once to prevent the deployer from changing the parameters after the fact.
-     * e) For safety, a parameters should be set before depositing any synthetic tokens in a liquidity pool.
-     * f) financialProduct must expose an expirationTimestamp method to validate it is correctly deployed.
+     * d) parameters can only be set once to prevent the deployer from changing the parameters after the fact.
+     * e) For safety, parameters should be set before depositing any synthetic tokens in a liquidity pool.
+     * f) longShortPair must expose an expirationTimestamp method to validate it is correctly deployed.
      */
     function setLongShortPairParameters(
-        address LongShortPair,
+        address longShortPair,
         int256 upperBound,
         int256 lowerBound
     ) public nonReentrant() {
-        require(ExpiringContractInterface(LongShortPair).expirationTimestamp() != 0, "Invalid LSP address");
+        require(ExpiringContractInterface(longShortPair).expirationTimestamp() != 0, "Invalid LSP address");
         require(upperBound > lowerBound, "Invalid bounds");
 
-        LinearLongShortPairParameters memory params = longShortPairParameters[LongShortPair];
+        LinearLongShortPairParameters memory params = longShortPairParameters[longShortPair];
         require(params.upperBound == 0 && params.lowerBound == 0, "Parameters already set");
 
-        longShortPairParameters[LongShortPair] = LinearLongShortPairParameters({
+        longShortPairParameters[longShortPair] = LinearLongShortPairParameters({
             upperBound: upperBound,
             lowerBound: lowerBound
         });
     }
 
     /**
-     * @notice Returns a number between 0 and 1 to indicate how much collateral each long and short token are entitled
+     * @notice Returns a number between 0 and 1e18 to indicate how much collateral each long and short token is entitled
      * to per collateralPerPair.
      * @param expiryPrice price from the optimistic oracle for the LSP price identifier.
      * @return expiryPercentLong to indicate how much collateral should be sent between long and short tokens.
      */
-    function computeExpiryTokensForCollateral(int256 expiryPrice)
+    function percentageLongCollateralAtExpiry(int256 expiryPrice)
         public
         view
         override
         nonReentrantView()
         returns (uint256)
     {
         LinearLongShortPairParameters memory params = longShortPairParameters[msg.sender];
+        require(params.upperBound != 0 && params.lowerBound != 0, "Params not set for calling LSP");
```

Comment:
Technically, the `lowerBound` can be 0, as long as the `upperBound` is > 0 right? The correct check should be `!(params.upperBound == 0 && params.lowerBound == 0)` I think.

Reply:
@nicholaspai your require is the same as the implementation, no?

@mrice32 that prevents the lower bound from being 0 which is a valid state!

Reply:
However, I'm happy to go with your syntax @nicholaspai. I've updated accordingly.

Reply:
@chrismaree your require would return false if the lowerBound = 0 which is a valid state. 

However I suppose this is a non-problem because somewhere else you require upperBound > lowerBound. 

Wdyt

Reply:
nope, it would not. the way it's set up now would return true if `lowerBound=0`, assuming `upperBound !=0`. The expression (like Matt pointed out) `params.upperBound != 0 || params.lowerBound != 0` is equivalent to what I had before which is `params.upperBound != 0 && params.lowerBound != 0` which basically will _only_ return false if both are 0. 

If the lower bound is 0 it's fine assuming the upper bound is not also 0.

do you agree?